### DISCUSSION TITLE
Alert Check Frequency

### DISCUSSION THREAD

**Discussion by nechon88** [datalust/seq-tickets#1721] [General]
How does Seq decide when it's time to "check" an alert? I see some alerts checked very frequently like every minute, and some checked every 15 minutes or so. Is it related to the time grouping or to the suppression time of the alert maybe?

**Comment by nblumhardt** [comment]
Hi! The algorithm varies a bit, but in general it will be based on the time grouping of the alert. Seq will try to check at around 1/3 the duration of the measurement window, within a minimum and maximum frequency.

Hope this helps,
Nick

**Comment by per-samuelsson** [comment]
I'm trying to get a better understanding of alerts and how to best use them, so I'm also interested in the topic on how they are implemented.

I'm trying to grasp such a simple thing as to how alerts are not firing multiple times on events in the stream. 

Suppose I log something, say once a minute, and set up an altert that would catch up to those logs, and my time grouping is say 6 seconds. Since the alert runs quite eagerly - every other-ish second then I guess, based on your writing above - what prevents it from sending an alert multiple times for the same log record? Is it as simple as that every alert filter with a start time that equal the end time from the previous run? 

(I didn't find any such details in the docs)

**Reply by nblumhardt** [reply]
Hi Per; yes, that's essentially how they work. Once an alert has fired in a given time range, no later checks will consider that time range.

Hope this helps,
Nick

**Reply by per-samuelsson** [reply]
Thanks @nblumhardt, that helps a lot. Really appreciate the fast and accurate answer! 👍
